src/ILCompiler/src/Program.cs

```diff
@@ -465,14 +470,28 @@ private int Run(string[] args)
             if (!_isCppCodegen && !_isWasmCodegen)
                 builder.UsePInvokePolicy(new ConfigurablePInvokePolicy(typeSystemContext.Target));
 
+            RemovedFeature removedFeatures = 0;
+            foreach (string feature in _removedFeatures)
+            {
+                if (feature == "EventSource")
+                    removedFeatures |= RemovedFeature.Etw;
+                else if (feature == "FrameworkStrings")
+                    removedFeatures |= RemovedFeature.FrameworkResources;
+            }
+
+            ILProvider ilProvider = null;
+            if (removedFeatures != 0)
+                ilProvider = new RemovingILProvider(new CoreRTILProvider(), removedFeatures);
+
             var stackTracePolicy = _emitStackTraceData ?
                 (StackTraceEmissionPolicy)new EcmaMethodStackTraceEmissionPolicy() : new NoStackTraceEmissionPolicy();
 
             MetadataBlockingPolicy mdBlockingPolicy = _noMetadataBlocking 
                     ? (MetadataBlockingPolicy)new NoMetadataBlockingPolicy() 
                     : new BlockedInternalsBlockingPolicy(typeSystemContext);
 
-            ManifestResourceBlockingPolicy resBlockingPolicy = new NoManifestResourceBlockingPolicy();
+            ManifestResourceBlockingPolicy resBlockingPolicy = (removedFeatures & RemovedFeature.FrameworkResources) != 0 ?
+                new FrameworkStringResourceBlockingPolicy() : (ManifestResourceBlockingPolicy)new NoManifestResourceBlockingPolicy();
```

Comment:
This also checks `IsFrameworkAssembly`. Should we check or assert it in the IL provider as well to make sure that the two are in sync?